public/scripts/main.js

```diff
@@ -1,43 +1,149 @@
 'use strict'
 
 const socket = require('socket.io-client')()
-var username = prompt('enter your name')
-var sound1 = document.getElementById('mySound')
-socket.on('incomingMessage', appendIncomingMessage)
-
-document.querySelector('#input').addEventListener('input', function(event) {
-  var nonLetters = /[^A-Z a-z]/
-  if (this.value.match(nonLetters)) {
-    sound1.currentTime = 0
-    sound1.play()
-    // need to add visual alert in to hud in future
-  }
-
-  this.value = this.value.toUpperCase()
-  this.value = this.value.replace(/[^A-Z ]/g, '')
-})
-
-function appendIncomingMessage(newMessage) {
-  var item = document.createElement('li')
-  item.appendChild(document.createTextNode(`<${newMessage.name}> ${newMessage.message}`))
-  document.getElementById('messages').appendChild(item)
-}
 
-// sends message to the server/terminal
-function sendMessage() {
-  var inputText = document.getElementById('input').value
-  document.getElementById('input').value = ''
+const ONE_SECOND = 1000
+
+// Initialize StateManager
+const stateManager = require('./stateManager')
+socket.on('stateChange', changeState)
 
-  socket.emit('message', {
-    name: username,
-    message: inputText
-  })
+function changeState(stateName) {
+  stateManager.changeTo(stateName)
+}
+
+// Shared Helper Functions
+function enterKeyPressed(callback) {
+  return function keyupEventListener(event) {
+    event.preventDefault()
+    const ENTER_KEY = 13
+    if (event.keyCode === ENTER_KEY) {
+      callback()
+    }
+  }
 }
 
-document.getElementById('input').addEventListener('keyup', function(event) {
-  event.preventDefault()
-  var ENTER_KEY = 13
-  if (event.keyCode === ENTER_KEY) {
-    sendMessage()
+// Login State
+const loginHandler = (function loginIIFE() {
+  let username
+  const nameInput = document.querySelector('#nameInput')
+  const registration = document.querySelector('#registration')
+  const lobby = document.querySelector('#lobby')
+  const lobbyList = document.querySelector('#lobbyList')
+  const countdownBanner = document.querySelector('#countdownBanner')
+
+  let timeoutReference
+
+  // Event Listeners
+  function updateLobby(usernames) {
+    lobbyList.innerHTML = ''
+    usernames.forEach(name => {
+      const listItem = document.createElement('li')
+      const text = (username === name) ? `> ${name}` : name
```

Comment:
I told you variable names are hard... `username` refers to this particular user's username, `usernames` refers to the list of all users that the server sent us. And `name` refers to each name in that usernames list as we iterate over it. If the current name in the list I'm looking at is my name, add a special style to it basically. Maybe I can make the naming clearer